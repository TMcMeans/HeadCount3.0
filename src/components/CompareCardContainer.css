.compare-container {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(250px, 1fr));
  grid-auto-rows: auto;
  grid-gap: 1rem;
  width: 85%;
  margin: 0 auto 50px auto;
  padding-bottom: 10px;
}
