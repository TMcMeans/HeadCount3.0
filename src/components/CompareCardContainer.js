import React from 'react';
import './CompareCardContainer.css';
import './CompareCard.css';

import { CompareCard } from './CompareCard';

export const CompareCardContainer = ({
  compareDistrictAverages,
  compareSchoolSelections,
  findAllMatches,
  removeCompareSelections
}) => {
  if (compareSchoolSelections.length === 1) {
    const schoolData = findAllMatches(compareSchoolSelections[0]);
    return (
      <div className="compare-container">
        <CompareCard
          className="compare-card"
          location={schoolData[0].location}
          stats={schoolData[0].stats}
          key={Math.random()}
          removeCompareSelections={removeCompareSelections}
        />
      </div>
    );
  } else if (compareSchoolSelections.length === 2) {
    const schoolData1 = findAllMatches(compareSchoolSelections[0]);
    const schoolData2 = findAllMatches(compareSchoolSelections[1]);
    const comparedData = compareDistrictAverages(
      compareSchoolSelections[0],
      compareSchoolSelections[1]
    );
    const comparedDataEntries = Object.entries(comparedData);
    if (comparedDataEntries.length > 2) {
      return (
        <div className="compare-container">
          <CompareCard
            className="compare-card"
            location={schoolData1[0].location}
            stats={schoolData1[0].stats}
            key={Math.random()}
            removeCompareSelections={removeCompareSelections}
          />
          <div className="comparison-section">
            <h1>COMPARE AVERAGES</h1>
            <h1>
              {comparedDataEntries[0][0]}: {comparedDataEntries[0][1]}
            </h1>
            <h1>
              COMPARED: <br />
              ---- {comparedDataEntries[2][1]} ----
            </h1>
            <h1>
              {comparedDataEntries[1][0]}: {comparedDataEntries[1][1]}
            </h1>
          </div>
          <CompareCard
            className="compare-card"
            location={schoolData2[0].location}
            stats={schoolData2[0].stats}
            key={Math.random()}
            removeCompareSelections={removeCompareSelections}
          />
        </div>
      );
    } else {
      return (
        <div className="compare-container">
          <h1>
            You must select two DIFFERENT school cards to compare averages.
          </h1>
        </div>
      );
    }
  } else {
    return <div className="compare-container" />;
  }
};
