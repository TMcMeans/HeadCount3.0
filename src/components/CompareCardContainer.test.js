import React from 'react';
import { shallow } from 'enzyme';
import kinderData from '../data/kindergartners_in_full_day_program.js';

import { CompareCardContainer } from './CompareCardContainer';
import DistrictRepository from '../helper.js';

describe('Compare Card Container', () => {
  it('should match snapshot with all html elements rendered with correct data', () => {
    const repository = new DistrictRepository(kinderData);
    const mockCompareSelections = ['ADAMS COUNTY 14', 'COLORADO'];
    const mockFunc = jest.fn();
    const wrapper = shallow(
      <CompareCardContainer
        compareDistrictAverages={repository.compareDistrictAverages}
        compareSchoolSelections={mockCompareSelections}
        findAllMatches={repository.findAllMatches}
        removeCompareSelections={mockFunc}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
