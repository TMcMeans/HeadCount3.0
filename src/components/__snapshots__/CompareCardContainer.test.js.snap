// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compare Card Container should match snapshot with all html elements rendered with correct data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    compareDistrictAverages={[Function]}
    compareSchoolSelections={
        Array [
            "ADAMS COUNTY 14",
            "COLORADO",
          ]
    }
    findAllMatches={[Function]}
    removeCompareSelections={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          className="compare-card"
          location="ADAMS COUNTY 14"
          removeCompareSelections={[Function]}
          stats={
                    Object {
                              "2004": 0.228,
                              "2005": 0.3,
                              "2006": 0.293,
                              "2007": 0.306,
                              "2008": 0.673,
                              "2009": 1,
                              "2010": 1,
                              "2011": 1,
                              "2012": 1,
                              "2013": 0.998,
                              "2014": 1,
                            }
          }
/>,
        <div
          className="comparison-section"
>
          <h1>
                    COMPARE AVERAGES
          </h1>
          <h1>
                    ADAMS COUNTY 14
                    : 
                    0.709
          </h1>
          <h1>
                    COMPARED: 
                    <br />
                    ---- 
                    1.338
                     ----
          </h1>
          <h1>
                    COLORADO
                    : 
                    0.53
          </h1>
</div>,
        <Unknown
          className="compare-card"
          location="COLORADO"
          removeCompareSelections={[Function]}
          stats={
                    Object {
                              "2004": 0.24,
                              "2005": 0.278,
                              "2006": 0.337,
                              "2007": 0.395,
                              "2008": 0.536,
                              "2009": 0.598,
                              "2010": 0.64,
                              "2011": 0.672,
                              "2012": 0.695,
                              "2013": 0.703,
                              "2014": 0.741,
                            }
          }
/>,
      ],
      "className": "compare-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0.1547781751800268",
        "nodeType": "function",
        "props": Object {
          "className": "compare-card",
          "location": "ADAMS COUNTY 14",
          "removeCompareSelections": [Function],
          "stats": Object {
            "2004": 0.228,
            "2005": 0.3,
            "2006": 0.293,
            "2007": 0.306,
            "2008": 0.673,
            "2009": 1,
            "2010": 1,
            "2011": 1,
            "2012": 1,
            "2013": 0.998,
            "2014": 1,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              COMPARE AVERAGES
</h1>,
            <h1>
              ADAMS COUNTY 14
              : 
              0.709
</h1>,
            <h1>
              COMPARED: 
              <br />
              ---- 
              1.338
               ----
</h1>,
            <h1>
              COLORADO
              : 
              0.53
</h1>,
          ],
          "className": "comparison-section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "COMPARE AVERAGES",
            },
            "ref": null,
            "rendered": "COMPARE AVERAGES",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "ADAMS COUNTY 14",
                ": ",
                0.709,
              ],
            },
            "ref": null,
            "rendered": Array [
              "ADAMS COUNTY 14",
              ": ",
              0.709,
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "COMPARED: ",
                <br />,
                "---- ",
                1.338,
                " ----",
              ],
            },
            "ref": null,
            "rendered": Array [
              "COMPARED: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              "---- ",
              1.338,
              " ----",
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "COLORADO",
                ": ",
                0.53,
              ],
            },
            "ref": null,
            "rendered": Array [
              "COLORADO",
              ": ",
              0.53,
            ],
            "type": "h1",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "0.735779932540203",
        "nodeType": "function",
        "props": Object {
          "className": "compare-card",
          "location": "COLORADO",
          "removeCompareSelections": [Function],
          "stats": Object {
            "2004": 0.24,
            "2005": 0.278,
            "2006": 0.337,
            "2007": 0.395,
            "2008": 0.536,
            "2009": 0.598,
            "2010": 0.64,
            "2011": 0.672,
            "2012": 0.695,
            "2013": 0.703,
            "2014": 0.741,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            className="compare-card"
            location="ADAMS COUNTY 14"
            removeCompareSelections={[Function]}
            stats={
                        Object {
                                    "2004": 0.228,
                                    "2005": 0.3,
                                    "2006": 0.293,
                                    "2007": 0.306,
                                    "2008": 0.673,
                                    "2009": 1,
                                    "2010": 1,
                                    "2011": 1,
                                    "2012": 1,
                                    "2013": 0.998,
                                    "2014": 1,
                                  }
            }
/>,
          <div
            className="comparison-section"
>
            <h1>
                        COMPARE AVERAGES
            </h1>
            <h1>
                        ADAMS COUNTY 14
                        : 
                        0.709
            </h1>
            <h1>
                        COMPARED: 
                        <br />
                        ---- 
                        1.338
                         ----
            </h1>
            <h1>
                        COLORADO
                        : 
                        0.53
            </h1>
</div>,
          <Unknown
            className="compare-card"
            location="COLORADO"
            removeCompareSelections={[Function]}
            stats={
                        Object {
                                    "2004": 0.24,
                                    "2005": 0.278,
                                    "2006": 0.337,
                                    "2007": 0.395,
                                    "2008": 0.536,
                                    "2009": 0.598,
                                    "2010": 0.64,
                                    "2011": 0.672,
                                    "2012": 0.695,
                                    "2013": 0.703,
                                    "2014": 0.741,
                                  }
            }
/>,
        ],
        "className": "compare-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0.1547781751800268",
          "nodeType": "function",
          "props": Object {
            "className": "compare-card",
            "location": "ADAMS COUNTY 14",
            "removeCompareSelections": [Function],
            "stats": Object {
              "2004": 0.228,
              "2005": 0.3,
              "2006": 0.293,
              "2007": 0.306,
              "2008": 0.673,
              "2009": 1,
              "2010": 1,
              "2011": 1,
              "2012": 1,
              "2013": 0.998,
              "2014": 1,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                COMPARE AVERAGES
</h1>,
              <h1>
                ADAMS COUNTY 14
                : 
                0.709
</h1>,
              <h1>
                COMPARED: 
                <br />
                ---- 
                1.338
                 ----
</h1>,
              <h1>
                COLORADO
                : 
                0.53
</h1>,
            ],
            "className": "comparison-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "COMPARE AVERAGES",
              },
              "ref": null,
              "rendered": "COMPARE AVERAGES",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "ADAMS COUNTY 14",
                  ": ",
                  0.709,
                ],
              },
              "ref": null,
              "rendered": Array [
                "ADAMS COUNTY 14",
                ": ",
                0.709,
              ],
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "COMPARED: ",
                  <br />,
                  "---- ",
                  1.338,
                  " ----",
                ],
              },
              "ref": null,
              "rendered": Array [
                "COMPARED: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                "---- ",
                1.338,
                " ----",
              ],
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "COLORADO",
                  ": ",
                  0.53,
                ],
              },
              "ref": null,
              "rendered": Array [
                "COLORADO",
                ": ",
                0.53,
              ],
              "type": "h1",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "0.735779932540203",
          "nodeType": "function",
          "props": Object {
            "className": "compare-card",
            "location": "COLORADO",
            "removeCompareSelections": [Function],
            "stats": Object {
              "2004": 0.24,
              "2005": 0.278,
              "2006": 0.337,
              "2007": 0.395,
              "2008": 0.536,
              "2009": 0.598,
              "2010": 0.64,
              "2011": 0.672,
              "2012": 0.695,
              "2013": 0.703,
              "2014": 0.741,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
